datafusion/expr/src/expr.rs

```diff
@@ -1574,40 +1600,38 @@ fn create_name(e: &Expr) -> Result<String> {
             Ok(parts.join(" "))
         }
         Expr::AggregateFunction(AggregateFunction {
-            fun,
+            func_def,
             distinct,
             args,
             filter,
             order_by,
-        }) => {
-            let mut name = create_function_name(&fun.to_string(), *distinct, args)?;
-            if let Some(fe) = filter {
-                name = format!("{name} FILTER (WHERE {fe})");
-            };
-            if let Some(order_by) = order_by {
-                name = format!("{name} ORDER BY [{}]", expr_vec_fmt!(order_by));
-            };
-            Ok(name)
-        }
-        Expr::AggregateUDF(AggregateUDF {
-            fun,
-            args,
-            filter,
-            order_by,
-        }) => {
-            let mut names = Vec::with_capacity(args.len());
-            for e in args {
-                names.push(create_name(e)?);
-            }
-            let mut info = String::new();
-            if let Some(fe) = filter {
-                info += &format!(" FILTER (WHERE {fe})");
+        }) => match func_def {
+            AggregateFunctionDefinition::BuiltIn { fun: _, name: _ }
+            | AggregateFunctionDefinition::Name(_) => {
+                let mut name = create_function_name(func_def.name(), *distinct, args)?;
```

Comment:
You might be able to combine the arms together here (the `AggregateFunctionDefinition::UDF` appears to do the same thing, though I realize it is not entirely the same)